http/task_service.go

```diff
@@ -1580,6 +1590,59 @@ func (t TaskService) FindRuns(ctx context.Context, filter influxdb.RunFilter) ([
 		return nil, 0, err
 	}
 
+	//handles usage of after and before time flags to filter runs output
+	if filter.BeforeTime != "" || filter.AfterTime != "" {
+		var timeStampBefore time.Time
+		var err error
+		if filter.BeforeTime != "" {
+			timeStampBefore, err = time.Parse(time.RFC3339, filter.BeforeTime)
+			if err != nil {
+				return nil, 0, err
+			}
+		}
+
+		var timeStampAfter time.Time
+		if filter.AfterTime != "" {
+			timeStampAfter, err = time.Parse(time.RFC3339, filter.AfterTime)
+			if err != nil {
+				return nil, 0, err
+			}
+		}
+
+		var flagRuns []*influxdb.Run //potentially dangerous bc dynamic allocation as slice grows in run-time (consider refactor here)
+		var run httpRun
+		var temp time.Time
+		if filter.BeforeTime != "" && filter.AfterTime != "" {
+			for j := range rs.Runs {
+				run = rs.Runs[j].httpRun
+				temp = *run.ScheduledFor
+				if (timeStampAfter.Before(temp)) && (temp.Before(timeStampBefore)) {
+					flagRuns = append(flagRuns, convertRun(run))
+				}
+			}
+		} else if filter.BeforeTime != "" {
+			for j := range rs.Runs {
+				run = rs.Runs[j].httpRun
+				temp = *run.ScheduledFor
+				if temp.Before(timeStampBefore) {
+					flagRuns = append(flagRuns, convertRun(run))
+				}
+			}
+		} else if filter.AfterTime != "" {
+			for j := range rs.Runs {
+				run = rs.Runs[j].httpRun
+				temp = *run.ScheduledFor
+				if timeStampAfter.Before(temp) {
+					flagRuns = append(flagRuns, convertRun(run))
+				}
+			}
+		} else {
+			return nil, 0, errors.New("error finding task runs")
+		}
```

Comment:
I think this could do with some de-duplication. You could do:
```go
var before time.Time
if filter.BeforeTime != "" {
	var err error
	before, err = time.Parse(time.RFC3339, filter.BeforeTime)
	if err != nil {
		return nil, 0, err
	}
}

var after time.Time
if filter.AfterTime != "" {
	var err error
	after, err = time.Parse(time.RFC3339, filter.AfterTime)
	if err != nil {
		return nil, 0, err
	}
}

between := func(scheduled time.Time) bool {
	return (after.IsZero() || after.Before(scheduled)) && (before.IsZero() || scheduled.Before(before))
}

for j := range rs.Runs {
	if run := rs.Runs[j].httpRun; between(*run.ScheduledFor) {
		flagRuns = append(flagRuns, convertRun(run))
	}
}
```

Also see final comment on location of this fix.

Reply:
That boolean function sums up all of my code in 1 return statement. I love it!

Reply:
@GeorgeMac  Regarding moving the code to kv/task.go, it seems to me that we're doing filtering in the HTTP TaskService and the kv TaskService. The difference is that we're finding runs in the HTTP area and finding tasks in the kv area. Because my solution is intended to find task runs, I'm not totally sure it would make sense to transfer that solution to the kv TaskService. Of course, I may be mistaken in my understanding of your comment so please do correct me. -Thanks!

Reply:
Ah I see. You're absolutely correct that I was referring to Tasks and not Runs. My mistake, thanks for clarity!.

I would still however push this logic down into the service and filter in the `TaskService.findRuns` method:
https://github.com/influxdata/influxdb/blob/648ce11df35152aedcfb7bbcfc164cae38a1b9d0/kv/task.go#L983-L1018

The primary reason is to keep the `http` transport layer thin and concerned with decoding/encoding requests and responses and then delegating to services. Rather than having it handle any kind of filtering. To keep responsibilities clear.
Then a test case can be added to the `testing` package as well. That should hold true across each implementation for this behavior.

Reply:
Hey @GeorgeMac, trying to make the solution work in kv here. A few small things: Firstly, my approach is to have the solution in kv and then call into that from the FindRuns function in HTTP. However, I'm running into issues because I have to call the kv FindRuns with a Service method receiver, which I don't exactly have. Can you give me some tips please? -thx